Sort output from kdep tests

Baltoli opened the pull request:
This should fix the flakiness issue where these tests would
intermittently fail on some CI runs but not others.

daejunpark commented:
@cos please review and merge it.

cos commented on API/pom.xml:
```diff
@@ -30,4 +30,23 @@
+        </plugins>
+    </build>
+-->
+
```
we can remove this

daejunpark replied:
Done.

cos commented on API/src/main/java/org/kframework/Kapi.java:
```diff
@@ -0,0 +1,481 @@
+/**
+ * KRunAPI
+ */
+public class Kapi {
```
let's merge and we'll work together on this later

cos commented on java-backend/src/main/java/org/kframework/backend/java/builtins/BuiltinIOOperations.java:
```diff
@@ -23,51 +22,48 @@
     }
 
-    public Term open(StringToken term1, StringToken term2, TermContext termContext) {
+    public static Term open(StringToken term1, StringToken term2, TermContext termContext) {
```
it's great these become static

cos commented on java-backend/src/main/java/org/kframework/backend/java/kil/KItem.java:
```diff
@@ -295,20 +294,19 @@ public BitSet getChildrenDontCareRuleMaskForPosition(int i) {
         public KItemOperations(
                 Stage stage,
-                JavaExecutionOptions javaOptions,
+                boolean deterministicFunctions,
```
nice

cos commented on java-backend/src/main/java/org/kframework/backend/java/symbolic/ConjunctiveFormula.java:
```diff
@@ -623,6 +624,27 @@ public boolean checkUnsat() {
+        if (!leftHandSide.implies(rightHandSide, rightOnlyVariables)) {
+            return null;
+        }
+         */
```
is this work in progress? let's add an issue then

cos commented on java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java:
```diff
@@ -74,6 +70,15 @@ public SymbolicRewriter(GlobalContext global, KompileOptions kompileOptions, Jav
+    // org.kframework.krun.KRunFrontEnd#run
+    // org.kframework.krun.KRun#run
+    // org.kframework.krun.modes.KRunExecutionMode#execute
+    // org.kframework.backend.java.symbolic.InitializeRewriter.SymbolicRewriterGlue#execute
```
add a comment above that this is the current stack trace

daejunpark replied:
Let me remove this, since now it's easy to find this call stack using IDE.

cos commented on kernel/src/main/java/org/kframework/kast/KastFrontEnd.java:
```diff
@@ -33,43 +25,28 @@
             FileUtil files,
-            @KompiledDir Provider<File> kompiledDir,
-            Provider<CompiledDefinition> compiledDef,
-            DefinitionScope scope) {
```
nice

cos commented on kernel/src/main/java/org/kframework/kompile/Kompile.java:
```diff
@@ -130,30 +137,32 @@ public Definition parseDefinition(File definitionFile, String mainModuleName, St
+            d = AddImplicitComputationCell.transformDefinition(d);
+            d = new Strategy(kompileOptions.experimental.heatCoolStrategies).addStrategyCellToRulesTransformer().apply(d);
+            d = ConcretizeCells.transformDefinition(d);
+            d = addSemanticsModule(d);
```
yep, easier to debug this way

cos commented on shell/pom.xml:
```diff
@@ -0,0 +1,57 @@
+    <artifactId>parent</artifactId>
+    <version>4.0.1-SNAPSHOT</version>
+  </parent>
+  <artifactId>shell</artifactId>
```
so this replaces the dependency injection. good.

daejunpark replied:
Yes, after removing dependency injection we had a circular dependency between modules, so I created another top layer to remove the circularity.

daejunpark commented:
@cos I made the requested change. Please review again and merge it if you are happy with that.

Baltoli committed:
Sort output from kdep tests

This should fix the flakiness issue where these tests would
intermittently fail on some CI runs but not others.

radumereuta commented on k-distribution/include/kframework/ktest-kdep.mak:
```diff
@@ -14,4 +14,4 @@ all: $(TESTS)
 
 %.k %.md: dummy
-	$(KDEP) $(OPTS) $@ | sed 's!'`pwd`'/\(\./\)\{0,2\}!!g' $(CHECK) $@.out
+	$(KDEP) $(OPTS) $@ | sed 's!'`pwd`'/\(\./\)\{0,2\}!!g' | sort $(CHECK) $@.out
```
1. Can we move this sort in the Java code? This would make it stable for other use cases in the future.
2. rename `OPTS` -> `KDEP_FLAGS` for consistency with other scripts.
3. Implement the `make clean` option. It doesn't do anything in this case, but it does return an error message and it stops the make clean from the parent directory. To be safe, copy the entire rule from ktest.mak.

Alternatively (for 2 and 3), could we include `ktest.mak`, but modify the local Makefile (like [here](https://github.com/kframework/k/blob/master/k-distribution/tests/regression-new/issue-2146-duplicateModules/Makefile)). If it's possible, I would prefer this option since it makes it clear it's a custom scenario and we don't need to create another .mak file.

Baltoli replied:
I've addressed these comments - as regards 2/3, I think the best way is to use a new `.mak` as Erik implemented it originally. The `ktest.mak` version does a lot of extra work (rather than just invoking kdep, which doesn't even need to `kompile` the source files). We already do this for `keq` tests as well.

Baltoli committed:
Revert "Sort output from kdep tests"

This reverts commit daed04273d70196d6ddf4647fe912aacb787d9bb.

Baltoli committed:
Sort in Java code, test style fixes

This commit moves the sorting of dependency files into the Java
frontend, rather than applying a sort to the test output files.

Additionally, it fixes stylistic problems with the testing code.

radumereuta reviewed (changes requested):
You need to update the outputs since they are now sorted.
```
regression-new/kdep-options$ make update-results
make: *** No rule to make target 'update-results'.  Stop.
```
Also, add this option to the new .mak file so we can update all the tests at once if we need to.

Baltoli committed:
Update sort order and results
